inspirehep/records/api/base.py

```diff
@@ -246,9 +246,9 @@ def get_subclasses(cls):
     @classmethod
     def get_record(cls, id_, with_deleted=False):
         record = super().get_record(str(id_), with_deleted)
-        type_from_schema = PidStoreBase.get_pid_type_from_schema(record["$schema"])
-        if record.pid_type is None or record.pid_type != type_from_schema:
-            return cls.get_subclasses()[type_from_schema].get_record(id_, with_deleted)
+        subclass = cls.get_class_from_schema(record)
```

Comment:
I don't like this naming but I don't have a better idea, ideas?

Reply:
`get_class_for_record` is better since you're passing the whole thing.